src/sonic_ax_impl/mibs/ietf/transceiver_sensor_data.py

```diff
@@ -0,0 +1,213 @@
+import re
+
+from .physical_entity_sub_oid_generator import SENSOR_TYPE_TEMP
+from .physical_entity_sub_oid_generator import SENSOR_TYPE_PORT_TX_POWER
+from .physical_entity_sub_oid_generator import SENSOR_TYPE_PORT_RX_POWER
+from .physical_entity_sub_oid_generator import SENSOR_TYPE_PORT_TX_BIAS
+from .physical_entity_sub_oid_generator import SENSOR_TYPE_VOLTAGE
+
+
+def transceiver_sensor_data():
+    """
+    Decorator for auto registering transceiver sensor data type
+    """
+    def wrapper(object_type):
+        TransceiverSensorData.register(object_type)
+        return object_type
+
+    return wrapper
+
+
+class TransceiverSensorData:
```

Comment:
>TransceiverSensorData [](start = 6, length = 21)

These classes seem over engineering. You don't use `self` much, so it is belonging to the same type with different fields.

Reply:
Hi @qiluo-msft , i am thinking how to change these classes to fields. One way might work could be using a class level dictionary to save different sensor data for different sensor types. Some code could be like:

```python
class TransceiverSensorData:
        sensor_data = {
        'temperature': {
            'name': 'Temperature',
            'oid_offset_base': SENSOR_TYPE_TEMP,
            'sort_factor': 0
        },
        'voltage': {
            'name': 'Voltage',
            'oid_offset_base': SENSOR_TYPE_VOLTAGE
            'sort_factor': 9000
        },
        'rx(\d+)power': {
            'name': 'RX Power',
            'oid_offset_base': SENSOR_TYPE_VOLTAGE,
            'sort_factor': 2000
        }
    }
```

Any suggestion?